### DISCUSSION TITLE
How to use useNavigate() in class

### DISCUSSION THREAD

**Discussion by renerlemes** [remix-run/remix#5554] [Q&A]
I have the following class:

```
import React from "react";

export default class Message extends React.Component {
	static Notificar() {
		navigate("/agencias");
	}
}
```

How to use navigate here?

**Comment by sergiodxa** [comment]
You can’t use hooks with classes. You need to create a HOC or Render Prop component being a function component and call the hook there, then pass it to the class component.

**Reply by renerlemes** [reply]
@sergiodxa Could you help me with an example in JS, based on my class?
